uui/components/navigation/MainMenu/MainMenuDropdown.tsx

```diff
@@ -1,53 +1,58 @@
-                    const handleEscape = (e: React.KeyboardEvent<HTMLElement>) => {
-                        if (e.key === 'Escape' && props.isOpen) {
-                            props.onClose();
-                        }
-                    };
+export function MainMenuDropdown(props: MainMenuDropdownProps) {
+    const handleEscape = (e: KeyboardEvent<HTMLElement>, onClose: () => void, isOpen: boolean) => {
+        if (e.key === 'Escape' && isOpen) {
+            onClose();
+        }
+    };
 
-                    return (
-                        <FocusLock returnFocus persistentFocus lockProps={ { onKeyDown: handleEscape } }>
-                            <div className={ cx(css.dropdownBody, 'uui-main_menu-dropdown', this.props.cx) }>
-                                {this.props.renderBody
-                                    ? this.props.renderBody({ ...props })
-                                    : React.Children.map<React.ReactElement<any>, React.ReactElement<any>>(this.props.children as React.ReactElement<any>[], (item) => {
-                                        if (!item) return item;
-                                        return React.createElement(item.type, {
-                                            ...item.props,
-                                            onClick: item.props.onClick
-                                                ? () => {
-                                                    item.props.onClick();
-                                                    props.onClose();
-                                                }
-                                                : null,
-                                        });
-                                    })}
-                            </div>
-                        </FocusLock>
-                    );
-                } }
-                placement="bottom-start"
-            />
-        );
-    }
+    return (
+        <Dropdown
+            renderTarget={ (dropdownProps) => (
+                <MainMenuButton
+                    caption={ props.caption }
+                    { ...dropdownProps }
+                    rawProps={ props.rawProps }
+                    isLinkActive={ props.isLinkActive }
+                    isDropdown
+                />
+            ) }
+            renderBody={ (dropdownBodyProps) => (
+                <FocusLock
+                    returnFocus
+                    persistentFocus
+                    lockProps={ {
+                        onKeyDown: (e: KeyboardEvent<HTMLElement>) => handleEscape(e, dropdownBodyProps.onClose, dropdownBodyProps.isOpen),
+                    } }
+                >
+                    <div className={ cx(css.dropdownBody, 'uui-main_menu-dropdown', props.cx) }>
+                        {props.renderBody
+                            ? props.renderBody({ ...dropdownBodyProps })
+                            : React.Children.map<ReactElement<any>, ReactElement<any>>(
```

Comment:
Probably we can also remove possibility to pass items for MainMenuDropdown via children?